### PR TITLE
enablePreflightReply feature added

### PR DESCRIPTION
**Pull request by logemann**
This feature (default -> true) handles all OPTION requests from a cors enabled browser automatically. Since a lot of people have problems understanding cors, i made it default to true. Only problem which might arise is if somebody really wants to handle other OPTION requests in their function. 

I have not found a way to distinguish a cors OPTION requests from potential business code based OPTION requests which are very rare anyway. Happy to discuss this scenario as well. 

Tell me what you think. Its a first try. Of course its tested in a real project of mine already.

### CODE REVIEW

**Comment by logemann** [comment]
i tested this against a very old version of middy core, which even wasnt called core back then but just "middy". I need to recheck things with current version.

File: packages/http-cors/index.js [hunk 1]
```diff
@@ -22,123 +69,53 @@ const defaults = {
...
-    if (existingHeaders.includes('Access-Control-Allow-Credentials')) {
-      options.credentials =
-        headers['Access-Control-Allow-Credentials'] === 'true'
+  const options = {
+    ...defaults,
+    ...opts
+  };
+  const httpCorsMiddlewareBefore = async (request) => {
+    if (options.enablePreflightReply) {
+      if (request.event.httpMethod === "OPTIONS") {
```
Comment (willfarrell) [inline packages/http-cors/index.js:87]: You should use the snippet below to ensure it works for all http event versions.
```
const httpMethod = getVersionHttpMethod[request.event.version ?? '1.0']?.(
      request.event
    )
    ```

File: packages/http-cors/index.js [hunk 2]
```diff
@@ -22,123 +69,53 @@ const defaults = {
   methods: undefined,
   origin: '*',
   origins: [],
+  enablePreflightReply: true,
```
Comment (willfarrell) [inline packages/http-cors/index.js:72]: What do you think of `disableEarlyPreflightResponse` or `disableStaticOrigins` (name need more thought, always the hardest part) set to `true`, with it changing to `false` as the default in the next major release. We use the disable notation in other middleware and have it disabled for now so it's not a breaking change.

File: packages/http-cors/index.js [hunk 3]
```diff
@@ -22,123 +69,53 @@ const defaults = {
...
   '1.0': (event) => event.httpMethod,
   '2.0': (event) => event.requestContext.http.method
-}
+};
+
+export default httpCorsMiddleware;
+
 
-export default httpCorsMiddleware
+//# sourceMappingURL=index.js.map
```
Comment (willfarrell) [inline packages/http-cors/index.js:121]: not needed

**Review by willfarrell — CHANGES_REQUESTED** [review]

**Comment by willfarrell** [comment]
Great start, can you also add in a unit test?

File: packages/http-cors/index.js (hunk 2 above)
Comment (logemann) [reply packages/http-cors/index.js:72]: i like disableEarlyPreflightResponse ... can we go with that?

**Comment by logemann** [comment]
quite busy this week. Will try to do the remaining changes hopefully this weekend.

**Comment by willfarrell** [comment]
No worries

**Pull request merged**